cf/irsa-api-role.yaml

```diff
@@ -81,22 +80,37 @@ Resources:
             Version: "2012-10-17"
             Statement:
               - Effect: Allow
-                Action: 's3:GetObject'
+                Action: "s3:GetObject"
                 Resource:
                   - !Sub "arn:aws:s3:::worker-results-${Environment}/*"
         - PolicyName: !Sub "can-access-details-of-redis-cluster-${Environment}"
           PolicyDocument:
             Version: "2012-10-17"
             Statement:
               - Effect: Allow
-                Action: 'elasticache:DescribeReplicationGroups'
+                Action: "elasticache:DescribeReplicationGroups"
                 Resource:
                   - !Sub "arn:aws:elasticache:${AWS::Region}:${AWS::AccountId}:replicationgroup:biomage-redis-${Environment}"
         - PolicyName: !Sub "can-access-details-of-kubernetes-cluster-${Environment}"
           PolicyDocument:
             Version: "2012-10-17"
             Statement:
               - Effect: Allow
-                Action: 'eks:DescribeCluster'
+                Action: "eks:DescribeCluster"
                 Resource:
                   - !Sub "arn:aws:eks:${AWS::Region}:${AWS::AccountId}:cluster/biomage-${Environment}"
+        - PolicyName: !Sub "can-manage-execution-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "states:DescribeStateMachine"
+                  - "states:CreateStateMachine"
+                  - "states:UpdateStateMachine"
+                  - "states:DescribeExecution"
+                  - "states:StartExecution"
+                  - "states:StopExecution"
+                Resource:
+                  - !Sub "arn:aws:states:${AWS::Region}:${AWS::AccountId}:stateMachine:biomage-pipeline-${Environment}-*"
+                  - !Sub "arn:aws:states:${AWS::Region}:${AWS::AccountId}:execution:biomage-pipeline-${Environment}-*:*"
```

Comment:
Even though this is supposedly valid, I think we should break up the state machine roles from the execution roles, and have them as separate policies.

That is, there should be a policy with `states:DescribeStateMachine` `states:CreateStateMachine` and `states:UpdateStateMachine` with the resource `!Sub "arn:aws:states:${AWS::Region}:${AWS::AccountId}:stateMachine:biomage-pipeline-${Environment}-*"`, and a separate one for the executions accordingly.

Other than that, lgtm.

Reply:
According the [AWS documentation](https://docs.aws.amazon.com/service-authorization/latest/reference/list_awsstepfunctions.html#awsstepfunctions-activity), the following permissions are available to state machines : 
 - `states:DescribeStateMachine`
  - `states:CreateStateMachine`
  - `states:UpdateStateMachine`
  - `states:StartExecution`

While these are available to executions
  - `states:DescribeExecution`
  - `states:StopExecution`
  
It's a bit weird to have `StartExecution` under `StateMachine`, but it makes sense